include/LeMonADE/feature/FeatureConnectionSc.h

```diff
@@ -0,0 +1,621 @@
+/*--------------------------------------------------------------------------------
+    ooo      L   attice-based  |
+  o\.|./o    e   xtensible     | LeMonADE: An Open Source Implementation of the
+ o\.\|/./o   Mon te-Carlo      |           Bond-Fluctuation-Model for Polymers
+oo---0---oo  A   lgorithm and  |
+ o/./|\.\o   D   evelopment    | Copyright (C) 2013-2015 by
+  o/.|.\o    E   nvironment    | LeMonADE Principal Developers (see AUTHORS)
+    ooo                        |
+----------------------------------------------------------------------------------
+
+This file is part of LeMonADE.
+
+LeMonADE is free software: you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation, either version 3 of the License, or
+(at your option) any later version.
+
+LeMonADE is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with LeMonADE.  If not, see <http://www.gnu.org/licenses/>.
+
+--------------------------------------------------------------------------------*/
+
+#ifndef LEMONADE_FEATURE_FEATURECONNECTIONSC_H
+#define LEMONADE_FEATURE_FEATURECONNECTIONSC_H
+
+#include <LeMonADE/feature/Feature.h>
+#include <LeMonADE/feature/FeatureExcludedVolumeSc.h>
+#include <LeMonADE/feature/FeatureLatticePowerOfTwo.h>
+#include <LeMonADE/updater/moves/MoveBase.h>
+#include <LeMonADE/updater/moves/MoveConnectBase.h>
+#include <LeMonADE/updater/moves/MoveConnectSc.h>
+#include <LeMonADE/updater/moves/MoveLocalBcc.h>
+#include <LeMonADE/updater/moves/MoveLocalSc.h>
+#include <LeMonADE/updater/moves/MoveAddMonomerBcc.h>
+#include <LeMonADE/updater/moves/MoveAddMonomerSc.h>
+#include <LeMonADE/utility/Lattice.h>
+#include <LeMonADE/io/AbstractRead.h>
+#include <LeMonADE/io/AbstractWrite.h>
+#include <LeMonADE/io/FileImport.h>
+/**
+ * @class MonomerReactivity
+ * @brief set the monomer reactive or unreactive 
+ * 
+ */
+class MonomerReactivity
+{
+  public:
+
+	//! Standard constructor- initially the reactivity is set to false.
+	MonomerReactivity():reactivity(false),nMaxBonds(2){}
+
+	//! Getting the reactivity of the monomer.
+	const bool isReactive() const {return reactivity;}
+	
+	//! Getting the number of maximum possible bonds for the monomer.
+	const uint32_t getNMaxBonds() const {return nMaxBonds;};
+
+	MonomerReactivity& operator= (const MonomerReactivity source)
+	{
+	  reactivity=source.isReactive();
+	  nMaxBonds=source.getNMaxBonds();
```

Comment:
There should be a check in the function to test against the "default" maximum connectivity for consistency reason.